shipyard_blueprints/microsoft-sharepoint/shipyard_microsoft_sharepoint/cli/download.py

```diff
@@ -0,0 +1,129 @@
+import os
+import sys
+import argparse
+import shipyard_bp_utils as shipyard
+from shipyard_templates import ShipyardLogger, ExitCodeException, CloudStorage
+from shipyard_microsoft_sharepoint import SharePointClient
+
+logger = ShipyardLogger.get_logger()
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--access-token", required=False, help="Access token to use for authentication"
+    )
+    parser.add_argument(
+        "--client-id", required=False, help="Client ID for OAuth authentication"
+    )
+    parser.add_argument(
+        "--client-secret", required=False, help="Client secret for OAuth authentication"
+    )
+    parser.add_argument(
+        "--tenant", required=False, help="Tenant for OAuth authentication"
+    )
+    parser.add_argument(
+        "--site-name",
+        required=True,
+        dest="site_name",
+        help="Site name to upload the file to",
+    )
+    parser.add_argument(
+        "--file-name", required=False, help="The name of the file once it is downloaded"
+    )
+    parser.add_argument("--directory", default="", required=False)
+    parser.add_argument(
+        "--sharepoint-file-name",
+        required=True,
+        default="",
+    )
+    parser.add_argument(
+        "--sharepoint-directory",
+        required=False,
+        default="",
+    )
+    parser.add_argument(
+        "--match-type",
+        required=False,
+        default="exact_match",
+        choices=["exact_match", "regex_match"],
+        help="Type of match to use when downloading the files",
+    )
+    return parser.parse_args()
+
+
+def main():
+    try:
+        args = get_args()
+        access_token = args.access_token
+        client_id = args.client_id
+        client_secret = args.client_secret
+        tenant = args.tenant
+        site_name = args.site_name
+
+        src_file = args.sharepoint_file_name
+        src_dir = args.sharepoint_directory
+        src_path = shipyard.files.combine_folder_and_file_name(src_dir, src_file)
+
+        target_file = args.file_name if args.file_name else src_file
+        target_dir = args.directory
+        target_path = shipyard.files.combine_folder_and_file_name(
+            target_dir, target_file
+        )
+
+        if target_dir:
+            shipyard.files.create_folder_if_dne(target_dir)
+
+        sharepoint = SharePointClient(
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant=tenant,
+            site_name=site_name,
+        )
+        sharepoint.connect()
+
+        if args.match_type == "exact_match":
```

Comment:
Not sure what specific change you are suggesting, I am using bp_utils where appropriate. I can't use the bp_utils to get the file matches for remove, download, and delete since those are all in the remote system, which is why I built the `get_file_matches` function in the class